.github/actions/system-info/action.yml

```diff
@@ -0,0 +1,24 @@
+name: system-info
+description: Get system information
+
+outputs:
+  platform:
+    description: The platform name
+    value: ${{ steps.system-info.outputs.platform }}
+  id:
+    description: The system ID
+    value: ${{ steps.system-info.outputs.id }}
+  os:
+    description: The OS name
+    value: ${{ steps.system-info.outputs.os }}
+  release:
+    description: The release version of the OS
+    value: ${{ steps.system-info.outputs.release }}
```

Comment:
A couple of comments:
1. only `os` and `release` seem to be used in workflows. Maybe `platform` and `id` could be added later when necessary.
2. `platform` (`linux`, `darwin`, `win32`) and `os` (`<distro>`, `macos`, `windows`) seem somehow mixed and redundant. Unless we want different caches for different distros (we only use ubuntu in the CI isnt' it?), why not using only `platform` for cache?

Reply:
1. Having those extra output cost nothing
2. I prefer to use `os` has you could run github action locally